internal/aws/k8scommon/k8sclient/clientset.go

```diff
@@ -0,0 +1,103 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+package k8sclient
+
+import (
+	"log"
+	"os"
+	"path/filepath"
+	"sync"
+	"time"
+
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+const (
+	cacheTTL = 10 * time.Minute
+)
+
+var client *K8sClient
+var Get func() *K8sClient
+
+func get() *K8sClient {
+	if !client.inited {
+		client.init()
+	}
+	return client
+}
+
+type K8sClient struct {
+	sync.Mutex
+	inited bool
+
+	ClientSet *kubernetes.Clientset
+
+	Ep   EpClient
```

Comment:
What is this client? Let's try to fix the comments on the code. It will be easier for the upstream PR review.

Reply:
Sure. I will add comments gradually.